skxray/diff_roi_choice.py

```diff
@@ -0,0 +1,354 @@
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+This module is to get informations of different region of interests(roi's).
+ Information : the number of pixels, pixel list, indices 
+"""
+
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+
+from six.moves import zip
+from six import string_types
+
+import time
+import sys
+import numpy as np
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def roi_rectangles(num_rois, roi_data, detector_size):
+    """
+    Parameters
+    ----------
+    num_rois: int
+        number of region of interests(roi)
+
+    roi_data: ndarray
+        upper left co-ordinates of roi's and the, length and width of roi's
+        from those co-ordinates
+        shape is [num_rois][4]
+
+    detector_size : tuple
+        2 element tuple defining the number of pixels in the detector.
+        Order is (num_rows, num_columns)
+
+    Returns
+    -------
+    roi_inds : ndarray
+        indices of the required shape
+        shape [detector_size[0]*detector_size[1]][1]
+
+    num_pixels : ndarray
+        number of pixels in certain rectangle shape
+
+    pixel_list : ndarray
+        pixel list for the required rings
+    """
+
+    mesh = np.zeros(detector_size, dtype=np.int64)
+
+    num_pixels = []
+
+    for i, (col_coor, row_coor, col_val, row_val) in enumerate(roi_data, 0):
+
+        left, right = np.max([col_coor, 0]), np.min([col_coor + col_val,
+                                                     detector_size[0]])
+        top, bottom = np.max([row_coor, 0]), np.min([row_coor + row_val,
+                                                     detector_size[1]])
+
+        area = (right - left) * (bottom - top)
+
+        # find the number of pixels in each roi
+        num_pixels.append(area)
+
+        slc1 = slice(left, right)
+        slc2 = slice(top, bottom)
+
+        # assign a different scalar for each roi
+        mesh[slc1, slc2] = (i + 1)
+
+    roi_inds = np.ravel(mesh)
+    pixel_list = np.where(roi_inds > 0)
+    roi_inds = roi_inds[roi_inds > 0]
+
+    return roi_inds, num_pixels, pixel_list
+
+
+def roi_rings(img_dim, calibrated_center, num_rings,
```

Comment:
I would implement this one by calling `roi_rings_step(...., np.ones(num_rings) * delta_r)`